src/views/home/components/amount-input/amount-input.view.tsx

```diff
@@ -58,14 +56,24 @@ const AmountInput: FC<AmountInputProps> = ({ value, token, balance, fee, onChang
   };
 
   const handleSendAll = () => {
-    const maxPossibleAmount = BigNumber.from(balance);
+    const maxPossibleAmount = balance;
     const maxAmountWithoutFee = maxPossibleAmount.sub(actualFee);
-    const newValue = getFixedTokenAmount(maxAmountWithoutFee.toString(), token.decimals);
 
-    setInputValue(newValue);
-    updateAmountInput(maxAmountWithoutFee);
+    if (maxAmountWithoutFee.gt(0)) {
+      const newValue = formatUnits(maxAmountWithoutFee, token.decimals);
+
+      setInputValue(newValue);
+      updateAmountInput(maxAmountWithoutFee);
+    }
   };
 
+  useEffect(() => {
```

Comment:
Why do we need this `useEffect` if this code is already in the two first lines of the component?

Reply:
Because we need to react if the value changes